psi4/src/CMakeLists.txt

```diff
@@ -1,6 +1,6 @@
 ### >> Create core target, so we may link modules to it directly <<
 ### >> This allow to propagate dependencies upwards sanely
-pybind11_add_module(core SHARED NO_EXTRAS core.cc)
```

Comment:
`SHARED NO_EXTRAS` might have been my doing at some point, but I don't recall why... This seems the correct way to go for this case, according to the latest pybind11 documentation.

Reply:
ok, I had supposed `SHARED NOEXTRAS` was from our last pb11 overhaul when we were still pushing back from plain `add_library(core ...)`. I'm would still have thought it should be `SHARED`, not `MODULE`, for the sake of the plugins. But plugins seem to work, and I don't think we want to link to `pybind11::embed` (https://github.com/pybind/pybind11/blob/master/tools/pybind11NewTools.cmake#L203-L207), so `MODULE` it is.

Reply:
As long as no one needs to do `-lpsi4` to produce _e.g._ an executable, we should use `MODULE`. I'm puzzled why it wasn't so already though :man_shrugging: